### PR TITLE
Optimization Form: allow user to select unfinalized items to include in the optimal schedule

### PR DESCRIPTION
**Pull request by hieudan225**
In this RQ, I added:
1)`OptimizationButton` to Trip.js
2) `OptimizationForm` which will pop up when user clicks into `OptimizationButton`
3) `DropArea` which will be used in `OptimizationForm`. This component allows user to drag and drop their unfinalized items (only event travelobject will be added to the list of selected items), calls the optimization functions when users click `Submit`.

For the purpose of drag-and-drop feature, I wrapped the `Unfinalized` and `OptimizationForm` in drag-and-drop-context, refactored some names to distinguish between `Finalized` and `Unfinalized` travelobject.

### CODE REVIEW

**Comment by hieudan225** [comment]
image

File: step-capstone/src/components/TravelObjects/DraggableUnfinalize.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import React, {useState} from "react";
+import { useDrag } from "react-dnd";
+import TravelObject from "./TravelObject";
+import _, { clone } from "lodash";
+import { ItemTypes } from "../../scripts/DragTravelObject";
+
+
+export default function DraggableUnfinalized(props) {
```
Comment (ganewman42) [inline step-capstone/src/components/TravelObjects/DraggableUnfinalize.js:8]: Nit: The function is `DraggableUnfinalized` but the file is named `DraggableUnfinalize`, please make the names match as in other files.

File: step-capstone/src/scripts/OptimizationForm.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+            position: "relative",
+            height: "850px",
+            overflowY: "scroll",
+            width: "600px",
+            margin: "0",
+          }}
+        >
+          <Grid item justify="center">
+            <Typography variant={"h5"} gutterBottom>
+              Optimization Form
```
Comment (ganewman42) [inline step-capstone/src/scripts/OptimizationForm.js:36]: Nit: Optimization Form is not my favorite name, maybe Schedule Optimization? Or Schedule Builder?

Comment (hieudan225) [reply step-capstone/src/scripts/OptimizationForm.js]: I have renamed it to ScheduleBuilder. Thanks for a great suggestion!

File: step-capstone/src/components/TravelObjects/DraggableUnfinalize.js (hunk 1 above)
Comment (hieudan225) [reply step-capstone/src/components/TravelObjects/DraggableUnfinalize.js]: I have renamed the file. Thanks!

**Pull request merged**